frontend/seo/product-list/hreflangs.ts

```diff
@@ -0,0 +1,260 @@
+   'iphone-4s': 'iphone-4s',
+   'ipad-2': 'ipad-2',
+   'iphone-5': 'iphone-5',
+   'ipad-mini-retina': 'ipad-mini-retina',
+   'imac-upgrade-kits': 'imac-upgrade-kits',
+   'apple-watch': 'apple-watch',
+   'iphone-5c': 'iphone-5c',
+   macbook: 'macbook',
+   'apple-macbook-parts': 'apple-macbook-parts',
+   parts: 'parts',
+};
+
+const auCaHreflangPaths: Record<Collection, string> = {
+   toolkits: 'toolkits',
+   'apple-iphone-parts': 'iphone',
+   'steam-deck-parts': 'steam-deck-parts',
+   'samsung-galaxy-parts': 'samsung-phone',
+   'macbook-pro': 'macbook-pro',
+   'nintendo-switch-parts': 'nintendo-switch',
+   'android-parts': 'android-parts',
+   'macbook-air': 'macbook-air',
+   'mac-parts': 'mac-parts',
+   'iphone-11': 'iphone-11',
+   'google-phone': 'google-phone',
+   'apple-ipad-parts': 'apple-ipad-parts',
+   'iphone-12-parts': 'iphone-12-parts',
+   'iphone-12-pro-max-parts': 'iphone-12-pro-max-parts',
+   'iphone-xr': 'iphone-xr',
+   'game-console-parts': 'game-console-parts',
+   'iphone-x': 'iphone-x',
+   'google-pixel-parts': 'google-pixel-parts',
+   'iphone-12-mini-parts': 'iphone-12-mini-parts',
+   'iphone-12-pro-parts': 'iphone-12-pro-parts',
+   'oneplus-parts': 'oneplus-phone',
+   'iphone-xs': 'iphone-xs',
+   'iphone-8': 'iphone-8',
+   'iphone-xs-max': 'iphone-xs-max',
+   'iphone-7': 'iphone-7',
+   'iphone-11-pro-max': 'iphone-11-pro-max',
+   'iphone-7-plus': 'iphone-7-plus',
+   'apple-imac-parts': 'apple-imac-parts',
+   'samsung-galaxy-s9': 'samsung-galaxy-s9',
+   'valve-index-parts': 'valve-index-parts',
+   'samsung-galaxy-s10-parts': 'samsung-galaxy-s10-parts',
+   'samsung-galaxy-s10-plus': 'samsung-galaxy-s10-plus',
+   'iphone-se': 'iphone-se',
+   'iphone-11-pro': 'iphone-11-pro',
+   'iphone-8-plus': 'iphone-8-plus',
+   'samsung-galaxy-s8': 'samsung-galaxy-s8',
+   'iphone-6s': 'iphone-6s',
+   'iphone-6': 'iphone-6',
+   'pc-laptop-parts': 'pc-laptop-parts',
+   'iphone-se-2020': 'iphone-se-2020',
+   'samsung-galaxy-s9-plus': 'samsung-galaxy-s9-plus',
+   'apple-mac-mini-parts': 'apple-mac-mini-parts',
+   'ipad-4': 'ipad-4',
+   'samsung-galaxy-s7': 'samsung-galaxy-s7',
+   'ipad-mini-1': 'ipad-mini-1',
+   'iphone-5s': 'iphone-5s',
+   'iphone-6s-plus': 'iphone-6s-plus',
+   'samsung-galaxy-s7-edge': 'samsung-galaxy-s7-edge',
+   'iphone-4': 'iphone-4',
+   'ipad-3': 'ipad-3',
+   'samsung-galaxy-s8-plus': 'samsung-galaxy-s8-plus',
+   'ipad-air-2': 'ipad-air-2',
+   'ipad-6': 'ipad-6',
+   'iphone-6-plus': 'iphone-6-plus',
+   'ipad-air-1': 'ipad-air-1',
+   'ipad-5': 'ipad-5',
+   'iphone-4s': 'iphone-4s',
+   'ipad-2': 'ipad-2',
+   'iphone-5': 'iphone-5',
+   'ipad-mini-retina': 'ipad-mini-retina',
+   'imac-upgrade-kits': 'imac-upgrade-kits',
+   'apple-watch': 'apple-watch',
+   'iphone-5c': 'iphone-5c',
+   macbook: 'macbook',
+   'apple-macbook-parts': 'mac-parts',
+   parts: 'parts',
+};
+
+const collections = Object.keys(
+   usHreflangsPaths
+) as unknown as Array<Collection>;
```

Comment:
I'm surprised this was needed.

Reply:
Apparently this is by design in TS https://github.com/Microsoft/TypeScript/issues/12870, `Object.keys()` returns `string[]` not `(keyof Obj)[]`